api/src/opentrons/protocol_engine/execution/equipment.py

```diff
@@ -319,14 +320,18 @@ def _find_applicable_labware_offset_id(
         if isinstance(labware_location, DeckSlotLocation):
             slot_name = labware_location.slotName
             module_model = None
-        else:
+        elif isinstance(labware_location, ModuleLocation):
             module_id = labware_location.moduleId
             # Allow ModuleNotLoadedError to propagate.
             module_model = self._state_store.modules.get_model(module_id=module_id)
             module_location = self._state_store.modules.get_location(
                 module_id=module_id
             )
             slot_name = module_location.slotName
+        else:
+            # No offset for off-deck location. This would allow loading a labware with
+            # 'off-deck' as valid location.
```

Comment:
```suggestion
            # No offset for off-deck location.
            # Returning None instead of raising an exception
            # allows loading a labware with 'off-deck' as valid location.
```

If I'm understanding this correctly?